sale_order_type/models/sale.py

```diff
@@ -73,6 +74,16 @@ def _prepare_invoice(self):
             res["sale_type_id"] = self.type_id.id
         return res
 
+    @api.constrains('company_id', 'type_id.company_id')
```

Comment:
```suggestion
    @api.constrains('company_id', 'type_id')
```
https://github.com/odoo/odoo/blob/c89b4c25ed92542f6fd0279cf3a9e4ef1bac9fa5/odoo/api.py#L119